andela/ah-infinity-stones
# #161966932 Bookmarks

#### What does this PR do?
- Integrates user ability to bookmark articles for later reading
- Allows article owner to view all bookmarks or delete a bookmark

#### Description of Task to be completed?
> User navigates to provided URLs or endpoints in the browser
> User should access the live application

#### How should this be manually tested?
`$ git clone -b ft-user-comments-161966918 https://github.com/andela/ah-infinity-stones`
`$ cd ah-infinity-stones`
> Create and activate a virtual envirinment to hold this app's dependancies
`$ virtualenv -p python3 venv`
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install dependencies`
> Setup environment variables in a file called `.env` with your own values. See sample below
> Load the `.env` variables by `$ source .env`
> Prepare migrations: `$ python3 manage.py makemigrations` and
> Run migrations: `$ python3 manage.py migrate`
> Run generic tests `$ python3 manage.py test`
> Run specific tests `$ python3 manage.py test authors/apps/comments`
> Register for a token to use on protected endpoint unless Read Only
> Engage the following endpoints:

> `POST http://127.0.0.1:8000/api/users/` to register
> `POST http://127.0.0.1:8000/api/articles/` to post article
> `POST http://127.0.0.1:8000/api/articles/<slug>/bookmark`
> `GET http://127.0.0.1:8000/api/bookmarks`
> `GET http://127.0.0.1:8000/api/bookmarks/?page=2`
> `DELETE http://127.0.0.1:8000/articles/<slug>/bookmark`

Populate your `.env` with:
```
export DATABASE_URL=postgres://<database_user>:<database_password>@localhost/<database_name>

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 export SECRET_KEY='your_secret'
```

#### What are the relevant pivotal tracker stories?
[#161966932](https://www.pivotaltracker.com/story/show/161966932)

#### Screenshots
<img width="1280" alt="screen shot 2019-01-04 at 17 26 43" src="https://user-images.githubusercontent.com/39657549/50694921-40122280-104c-11e9-8fce-042873482714.png">

<img width="1280" alt="screen shot 2019-01-04 at 17 31 42" src="https://user-images.githubusercontent.com/39657549/50694936-4b654e00-104c-11e9-98ee-9e49938f4ff5.png">

<img width="1280" alt="screen shot 2019-01-04 at 17 26 54" src="https://user-images.githubusercontent.com/39657549/50694932-47d1c700-104c-11e9-90d9-ef7ac5d6c872.png">

<img width="1280" alt="screen shot 2019-01-04 at 18 24 08" src="https://user-images.githubusercontent.com/39657549/50695511-f32f4b80-104d-11e9-88d9-717d63335445.png">

<img width="1280" alt="screen shot 2019-01-04 at 18 25 29" src="https://user-images.githubusercontent.com/39657549/50695599-28d43480-104e-11e9-897a-53058e74ac6f.png">
